### PR TITLE
[21159] Remove include/fastrtps

### PR DESCRIPTION
**Pull request by EduPonz**
## Description

This PR removes the `include/fastrtps` directory completely.

Depends on:

* #4898 

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- _N/A_: Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- _N/A_: Any new/modified methods have been properly documented using Doxygen. 
- _N/A_: Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- ❌ Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- ❌ Changes are API compatible. 
- _N/A_: New feature has been added to the `versions.md` file (if applicable).
- [x] New feature has been documented/Current behavior is correctly described in the documentation. 
    - Related documentation PR: eProsima/Fast-DDS-docs#810
- _N/A_: Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: include/fastdds/dds/core/policy/ParameterTypes.hpp [hunk 1]
```diff
@@ -1828,6 +1826,35 @@ T get_proxy_property(
 }
 
 } //namespace dds
+
+namespace rtps {
+
+using ParameterId_t = fastdds::dds::ParameterId_t;
```
Comment (elianalf) [inline include/fastdds/dds/core/policy/ParameterTypes.hpp:1832]: Do we actually need to keep these classes and enum in both namespaces or we can keep only `eprosima::fastdds::dds`?

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 2]
```diff
@@ -444,117 +448,121 @@ bool ReaderProxyData::writeToCDRMessage(
...
-        if (!fastdds::dds::QosPoliciesSerializer<DurabilityQosPolicy>::add_to_cdr_message(m_qos.m_durability, msg))
+        if (!fastdds::dds::QosPoliciesSerializer<dds::DurabilityQosPolicy>::add_to_cdr_message(m_qos.m_durability, msg))
         {
             return false;
         }
     }
     if (m_qos.m_durabilityService.send_always() || m_qos.m_durabilityService.hasChanged)
     {
-        if (!fastdds::dds::QosPoliciesSerializer<DurabilityServiceQosPolicy>::add_to_cdr_message(m_qos.
+        if (!fastdds::dds::QosPoliciesSerializer<dds::DurabilityServiceQosPolicy>::add_to_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:458]:
```suggestion
        if (!fastdds::dds::QosPoliciesSerializer<dds::DurabilityServiceQosPolicy>::add_to_cdr_message(
        m_qos.m_durabilityService, msg))
```

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 3]
```diff
@@ -444,117 +448,121 @@ bool ReaderProxyData::writeToCDRMessage(
...
-        if (!fastdds::dds::QosPoliciesSerializer<UserDataQosPolicy>::add_to_cdr_message(m_qos.m_userData, msg))
+        if (!fastdds::dds::QosPoliciesSerializer<dds::UserDataQosPolicy>::add_to_cdr_message(m_qos.m_userData, msg))
         {
             return false;
         }
     }
     if (m_qos.m_timeBasedFilter.send_always() || m_qos.m_timeBasedFilter.hasChanged)
     {
-        if (!fastdds::dds::QosPoliciesSerializer<TimeBasedFilterQosPolicy>::add_to_cdr_message(m_qos.m_timeBasedFilter,
+        if (!fastdds::dds::QosPoliciesSerializer<dds::TimeBasedFilterQosPolicy>::add_to_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:510]:
```suggestion
        if (!fastdds::dds::QosPoliciesSerializer<dds::TimeBasedFilterQosPolicy>::add_to_cdr_message(
        m_qos.m_timeBasedFilter, msg))
```

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 4]
```diff
@@ -444,117 +448,121 @@ bool ReaderProxyData::writeToCDRMessage(
...
-        if (!fastdds::dds::QosPoliciesSerializer<OwnershipQosPolicy>::add_to_cdr_message(m_qos.m_ownership, msg))
+        if (!fastdds::dds::QosPoliciesSerializer<dds::OwnershipQosPolicy>::add_to_cdr_message(m_qos.m_ownership, msg))
         {
             return false;
         }
     }
     if (m_qos.m_destinationOrder.send_always() || m_qos.m_destinationOrder.hasChanged)
     {
-        if (!fastdds::dds::QosPoliciesSerializer<DestinationOrderQosPolicy>::add_to_cdr_message(m_qos.m_destinationOrder,
+        if (!fastdds::dds::QosPoliciesSerializer<dds::DestinationOrderQosPolicy>::add_to_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:526]:
```suggestion
        if (!fastdds::dds::QosPoliciesSerializer<dds::DestinationOrderQosPolicy>::add_to_cdr_message(
        m_qos.m_destinationOrder,
```

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 5]
```diff
@@ -444,117 +448,121 @@ bool ReaderProxyData::writeToCDRMessage(
...
+        if (!fastdds::dds::QosPoliciesSerializer<dds::GroupDataQosPolicy>::add_to_cdr_message(m_qos.m_groupData, msg))
         {
             return false;
         }
     }
     if ((m_qos.m_disablePositiveACKs.send_always() || m_qos.m_topicData.hasChanged) &&
             m_qos.m_disablePositiveACKs.enabled)
     {
-        if (!fastdds::dds::QosPoliciesSerializer<DisablePositiveACKsQosPolicy>::add_to_cdr_message(m_qos.
+        if (!fastdds::dds::QosPoliciesSerializer<dds::DisablePositiveACKsQosPolicy>::add_to_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:565]:
```suggestion
        if (!fastdds::dds::QosPoliciesSerializer<dds::DisablePositiveACKsQosPolicy>::add_to_cdr_message(
        m_qos.m_disablePositiveACKs, msg))
```

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 6]
```diff
@@ -610,7 +619,8 @@ bool ReaderProxyData::writeToCDRMessage(
 
     if (m_qos.representation.send_always() || m_qos.representation.hasChanged)
     {
-        if (!fastdds::dds::QosPoliciesSerializer<DataRepresentationQosPolicy>::add_to_cdr_message(m_qos.representation,
+        if (!fastdds::dds::QosPoliciesSerializer<dds::DataRepresentationQosPolicy>::add_to_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:622]:
```suggestion
        if (!fastdds::dds::QosPoliciesSerializer<dds::DataRepresentationQosPolicy>::add_to_cdr_message(
        m_qos.representation, msg))
```

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 7]
```diff
@@ -619,7 +629,7 @@ bool ReaderProxyData::writeToCDRMessage(
 
     if (m_qos.type_consistency.send_always() || m_qos.type_consistency.hasChanged)
     {
-        if (!fastdds::dds::QosPoliciesSerializer<TypeConsistencyEnforcementQosPolicy>::add_to_cdr_message(m_qos.
+        if (!fastdds::dds::QosPoliciesSerializer<dds::TypeConsistencyEnforcementQosPolicy>::add_to_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:632]:
```suggestion
        if (!fastdds::dds::QosPoliciesSerializer<dds::TypeConsistencyEnforcementQosPolicy>::add_to_cdr_message(
        m_qos.type_consistency, msg))
```

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 8]
```diff
@@ -685,7 +695,7 @@ bool ReaderProxyData::readFromCDRMessage(
                     }
                     case fastdds::dds::PID_DEADLINE:
                     {
-                        if (!fastdds::dds::QosPoliciesSerializer<DeadlineQosPolicy>::read_from_cdr_message(m_qos.
+                        if (!fastdds::dds::QosPoliciesSerializer<dds::DeadlineQosPolicy>::read_from_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:698]:
```suggestion
                        if (!fastdds::dds::QosPoliciesSerializer<dds::DeadlineQosPolicy>::read_from_cdr_message(
                        m_qos.m_deadline, msg, plength))
```

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 9]
```diff
@@ -694,16 +704,17 @@ bool ReaderProxyData::readFromCDRMessage(
                     }
                     case fastdds::dds::PID_LATENCY_BUDGET:
                     {
-                        if (!fastdds::dds::QosPoliciesSerializer<LatencyBudgetQosPolicy>::read_from_cdr_message(m_qos.
-                                        m_latencyBudget, msg, plength))
+                        if (!fastdds::dds::QosPoliciesSerializer<dds::LatencyBudgetQosPolicy>::read_from_cdr_message(
+                                    m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:708]:
```suggestion
                                    m_qos.m_latencyBudget, msg, plength))
```

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 10]
```diff
@@ -721,7 +733,7 @@ bool ReaderProxyData::readFromCDRMessage(
                     }
                     case fastdds::dds::PID_LIFESPAN:
                     {
-                        if (!fastdds::dds::QosPoliciesSerializer<LifespanQosPolicy>::read_from_cdr_message(m_qos.
+                        if (!fastdds::dds::QosPoliciesSerializer<dds::LifespanQosPolicy>::read_from_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:736]:
```suggestion
                        if (!fastdds::dds::QosPoliciesSerializer<dds::LifespanQosPolicy>::read_from_cdr_message(
                        m_qos.m_lifespan, msg, plength))
```

File: src/cpp/rtps/builtin/data/ReaderProxyData.cpp [hunk 11]
```diff
@@ -730,7 +742,7 @@ bool ReaderProxyData::readFromCDRMessage(
                     }
                     case fastdds::dds::PID_USER_DATA:
                     {
-                        if (!fastdds::dds::QosPoliciesSerializer<UserDataQosPolicy>::read_from_cdr_message(m_qos.
+                        if (!fastdds::dds::QosPoliciesSerializer<dds::UserDataQosPolicy>::read_from_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/ReaderProxyData.cpp:745]: Please check the whole function, almost all of then are wrongly indented

File: src/cpp/rtps/builtin/data/WriterProxyData.cpp [hunk 12]
```diff
@@ -439,118 +443,122 @@ bool WriterProxyData::writeToCDRMessage(
...
-        if (!fastdds::dds::QosPoliciesSerializer<DurabilityQosPolicy>::add_to_cdr_message(m_qos.m_durability, msg))
+        if (!fastdds::dds::QosPoliciesSerializer<dds::DurabilityQosPolicy>::add_to_cdr_message(m_qos.m_durability, msg))
         {
             return false;
         }
     }
     if (m_qos.m_durabilityService.send_always() || m_qos.m_durabilityService.hasChanged)
     {
-        if (!fastdds::dds::QosPoliciesSerializer<DurabilityServiceQosPolicy>::add_to_cdr_message(m_qos.
+        if (!fastdds::dds::QosPoliciesSerializer<dds::DurabilityServiceQosPolicy>::add_to_cdr_message(m_qos.
```
Comment (elianalf) [inline src/cpp/rtps/builtin/data/WriterProxyData.cpp:453]: Can you please check also this function?
```suggestion
        if (!fastdds::dds::QosPoliciesSerializer<dds::DurabilityServiceQosPolicy>::add_to_cdr_message(
        m_qos.m_durabilityService, msg))
```

**Review by elianalf — CHANGES_REQUESTED** [review]

File: include/fastdds/dds/core/policy/ParameterTypes.hpp (hunk 1 above)
Comment (EduPonz) [reply include/fastdds/dds/core/policy/ParameterTypes.hpp:1832]: We decided to keep it as the Parameters are also an RTPS concept, it made sense

**Comment by EduPonz** [comment]
@richiprosima please test_3 discovery-server

**Review by elianalf — APPROVED** [review]

**Pull request merged**